misc/python/materialize/output_consistency/input_data/params/number_operation_param.py

```diff
@@ -24,6 +31,10 @@ def __init__(
         incompatibility_combinations: Optional[
             List[Set[ExpressionCharacteristics]]
         ] = None,
+        only_int_type: bool = False,
+        no_int_type_larger_int4: bool = False,
+        no_floating_point_type: bool = False,
+        no_unsigned_type: bool = False,
```

Comment:
I think I might have created a list of types and implemented these as a lambda/small function.

Reply:
Can you explain what the benefit would be?

Reply:
The supports_type method could then be a trivial operation, and each user of NumericOperationParam could define their own filter. But I guess it's ok as is.